### ISSUE TITLE
`TomlConfigParser` fails to handle environment variables whose value is a JSON-encoded string.

### ISSUE THREAD

**Issue by stevenhoelscher** [coveragepy/coveragepy#1345]
**Describe the bug**

`TomlConfigParser` fails to handle environment variables whose value is a JSON-encoded string.

**To Reproduce**
How can we reproduce the problem?  Please *be specific*. Don't link to a failing CI job. Answer the questions below:
1. What version of Python are you using? `3.10.2`
1. What version of coverage.py shows the problem? The output of `coverage debug sys` is helpful. `6.2`
1. What versions of what packages do you have installed? The output of `pip freeze` is helpful.

```
tomli==2.0.0
pytest==6.2.5
pytest-cov==3.0.0
```
1. What code shows the problem?  Give us a specific commit of a specific repo that we can check out. If you've already worked around the problem, please provide a commit before that fix.

See https://github.com/stevenhoelscher/coveragepy-repro for reproduction of the bug.

1. What commands did you run?
```
pytest
```

**Expected behavior**
When running `pytest` with `pytest-cov` (which relies on `coveragepy` under the hood), another tool's `pyproject.toml` should not affect `coveragepy`'s ability to parse the configuration file.

**Additional context**

As you can see in this example, [I try to add another tool's configuration](https://github.com/stevenhoelscher/coveragepy-repro/blob/main/pyproject.toml#L1-L3) to my `pyproject.toml`. But the [`substitute_variables`](https://github.com/nedbat/coveragepy/blob/master/coverage/tomlconfig.py#L49) function ends up interpolating the environment variable and invalidating the configuration prior to parsing:

```
'[[tool.docker.env_vars]]\nname = "foo"\nvalue = "\\"{"foo": "bar"}\\""\n\n[tool.pytest.ini_options]\ntestpaths = "src/tests"\naddopts = "--cov=src"\n\n[tool.coverage.run]\nomit = ["src/tests/*"]\n'
```

```
>>> import tomli
>>> s = '[[tool.docker.env_vars]]\nname = "foo"\nvalue = "\\"{"foo": "bar"}\\""\n\n[tool.pytest.ini_options]\ntestpaths = "src/tests"\naddopts = "--cov=src"\n\n[tool.coverage.run]\nomit = ["src/tests/*"]\n'
>>> tomli.loads(s)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/developer/code/coveragepy-repro/.venv/lib/python3.10/site-packages/tomli/_parser.py", line 116, in loads
    raise suffixed_err(
tomli.TOMLDecodeError: Expected newline or end of document after a statement (at line 3, column 14)
```

**Comment by nedbat** [comment]
I believe this is fixed as of commit 44fbd3b0.  Would you be able to try the latest coverage from master to see?
```
pip install git+https://github.com/nedbat/coveragepy@master
```

**Comment by nedbat** [comment]
This is fixed as of commit https://github.com/nedbat/coveragepy/commit/44fbd3b02ad22326767dc37fe3b94aa93b36e8a3.

**Comment by nedbat** [comment]
This is now released as part of [coverage 6.6.0b1](https://pypi.org/project/coverage/6.6.0b1).

**Comment by stevenhoelscher** [comment]
Works, thanks so much!

**Comment by nedbat** [comment]
This is now released as part of [coverage 7.0.0b1](https://pypi.org/project/coverage/7.0.0b1).
